core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -78,7 +78,7 @@ public OutputFile newOutputFile(String path) {
 
   @Override
   public void deleteFile(String path) {
-    Path toDelete = new Path(path);
+    Path toDelete = new Path(path.replaceAll("s3", "s3a"));
```

Comment:
In some situations, these are considered the same -- specifically in S3FileIO on read.

In other parts of the code, equivalency of schemes is something that can be configured.

Some people might  want more s3 prefixes to be considered, such as `s3n`. So I'm more of a fan of the second approach.

But I've been OOO for a bit, so might not be fully up to date. I would be interested to see what @flyrain thinks about this change to HadoopFileIO, as I believe he introduced the configuration for allowing different prefixes to be considered equivalent.

Reply:
This this does seem to be done in many places though, if it's not presently configurable in the general case, maybe we should add a function that normalizes based off of some configuration?

I think that the configurable equivalency check that @flyrain added was for file cleanup or some other procedure. So if it's not configurable at the catalog level or the file io level, maybe that would be best to consider?

Though admittedly this change makes me a bit nervous as we have file names hard coded in places.

That general hesitancy is why I'd like Yufei's input. I'm sure he had to deal with most cases when he implemented equivalency checking for certain procedures.

Reply:
Here's the equivalency configuration I was thinking of: https://github.com/apache/iceberg/blob/336d18acdb7dc0e783861151e2e871524cef9f6d/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java#L101-L105

Sorry I've been OOO recovering from a surgery 😅

Reply:
Also, how much if any of the equivalency checking can we defer to the various hadoop libraries (cloud shims, native HDFS) would be good to know as well.

But I can see cases where we'd want to be able to control it ourselves (cross cloud interactions for disaster recovery, etc).